### PR TITLE
Add explicit dependency between shop-pddl-tests and warns-check.lisp

### PR DESCRIPTION
**Pull request by matthewdehaven**
By moving warns-check.lisp into the shop-test-helper component, then adding a depends-on to shop-pddl-tests for that module. The other component which was depending on the warns-check, shop-internal-tests, already depends on the shop-test-helper component.

Addresses #138

### CODE REVIEW

**Comment by rpgoldman** [comment]
The version of `fiveam` that is included as SHOP submodule has `fiveam:warns` defined already.  I have a suggestion for a minor refinement that I will post in the review.

File: shop3/shop3.asd [hunk 1]
```diff
@@ -266,12 +266,14 @@ shop3."
     :version (:read-file-form "shop-version.lisp-expr")
     :components ((:module "shop-test-helper"
                           :pathname "tests/"
-                          :components ((:file "common")))
+                          :components ((:file "common")
+                                       (:file "warns-check")))
```
Comment (rpgoldman) [inline shop3/shop3.asd:270]: I think we should check to see if this macro is already defined by FiveAM (there's a [pull request](https://github.com/lispci/fiveam/pull/81) for it).  So at the top of warns-check, add:

``` lisp
(eval-when (:compile-toplevel :load-toplevel :execute)
    (when (fboundp (uiop:intern* '#:warns :fiveam)) (pushnew :fiveam-warns features)))

#+(not fiveam-warns)
<warns definition>
```
If I knew that this was going to be added at a particular FiveAM version, I could do this in a cleaner way, but 🤷‍♂️

**Review by rpgoldman — CHANGES_REQUESTED** [review]
If you could make that change and verify it, I will get this merged.

File: shop3/shop3.asd (hunk 1 above)
Comment (matthewdehaven) [reply shop3/shop3.asd:270]: I added the check before the warns definition here: https://github.com/shop-planner/shop3/pull/139/commits/ccb211dbd3116aa8d2e85f5fa79ab325f36423e1

Let me know if that's not what you are expecting and I can fix it.

Comment (matthewdehaven) [reply shop3/shop3.asd:270]: The CI tests are failing after adding the check for existence of warns. At a glance, the test failure looks unrelated to the change, so I’m confused at the moment.

I’ll run the tests locally later today to see if I can replicate and debug the test failure.

**Comment by rpgoldman** [comment]
@matthewdehaven  Checking `master` to see why the tests are failing here.  Merge to come soon, promise!

**Comment by rpgoldman** [comment]
I have found what's wrong. Somehow the call to `warns` is not returning the value of the enclosed form, as it should.  I will see if I can figure out why.

**Pull request merged**